src/main/java/org/kohsuke/groovy/sandbox/impl/GroovyCallSiteSelector.java

```diff
@@ -121,17 +121,34 @@ private static boolean isIllegalCallToSyntheticConstructor(Constructor<?> c, Obj
         Class<?>[] parameterTypes = c.getParameterTypes();
         for (int i = 0; i < parameterTypes.length; i++) {
             Class<?> parameterType = parameterTypes[i];
-            Object arg = args[i];
-            for (Class<?> syntheticParamType : SYNTHETIC_CONSTRUCTOR_PARAMETER_TYPES) {
-                // Using explicit equality checks instead of `instanceOf` to disallow subclasses of SuperConstructorWrapper and ThisConstructorWrapper.
-                if (parameterType == syntheticParamType && (arg == null || arg.getClass() != syntheticParamType)) {
+            if (c.isVarArgs() && i == parameterTypes.length - 1) {
+                // If this is a vararags constructor, there can be 0 or more arguments for the final parameter.
+                for (int j = i; j < args.length; j++) {
+                    Object arg = args[j];
+                    if (isIllegalArgumentForSyntheticParameter(parameterType, arg)) {
+                        return true;
+                    }
+                }
```

Comment:
Alternatively, maybe we should just check if the parameter type is a synthetic constructor parameter type, and if so, return `true`, otherwise return `false`, without checking all of the arguments, since we will never generate a synthetic constructor with a varargs argument that is one of the wrapper types, so if we run into a varargs `*ConstructorWrapper` parameter, something is wrong and the call should be rejected.

Reply:
Also, I think that `parameterType` will be an array in the cases we care about, so the current check is wrong, since we need to compare the arguments against the element type of the array.